src/Orchard.Cms.Web/Modules/Orchard.Roles/Recipes/RolesStep.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Orchard.Recipes.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Orchard.Recipes.Models;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Identity;
+using Orchard.Security;
+using Orchard.Security.Permissions;
+using Orchard.Environment.Extensions;
+
+namespace Orchard.Roles.Recipes
+{
+    public class RolesStep : RecipeExecutionStep
+    {
+        private readonly RoleManager<Role> _roleManager;
+        private readonly ITypeFeatureProvider _typeFeatureProvider;
+        private readonly IEnumerable<IPermissionProvider> _permissionProviders;
+
+
+        public RolesStep(RoleManager<Role> roleManager,
+                            ITypeFeatureProvider typeFeatureProvider,
+                            IEnumerable<IPermissionProvider> permissionProviders,
+                            ILoggerFactory loggerFactory, 
+                            IStringLocalizer<RolesStep> stringLocalizer) : base(loggerFactory, stringLocalizer)
+        {
+            _roleManager = roleManager;
+            _typeFeatureProvider = typeFeatureProvider;
+            _permissionProviders = permissionProviders;
+        }
+
+        public override string Name
+        {
+            get { return "Roles"; }
+        }
+
+        public override async Task ExecuteAsync(RecipeExecutionContext recipeContext)
+        {
+            var model = recipeContext.RecipeStep.Step.ToObject<RolesStepModel>();
+            var validPermissions = GetValidPermissions();
```

Comment:
No it doesn't fail. I check it to avid typos in recipe. I will remove that check